src/Traits/ManagesInvoices.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+
+namespace Laravel\Cashier\Traits;
+
+use Dompdf\Options;
+use Illuminate\Support\Str;
+use Laravel\Cashier\Cashier;
+use Laravel\Cashier\Exceptions\UnauthorizedInvoiceAccessException;
+use Laravel\Cashier\OneOffPayment\OneOffPaymentBuilder;
+use Laravel\Cashier\Order\Invoice;
+use Laravel\Cashier\Order\Order;
+use Laravel\Cashier\Order\OrderItem;
+use Laravel\Cashier\Order\OrderItemCollection;
+use Laravel\Cashier\SubscriptionBuilder\RedirectToCheckoutResponse;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+trait ManagesInvoices
+{
+    /**
+     * Find an invoice by ID.
+     *
+     * @param string $orderId
+     * @return Invoice|null
+     * @throws \Laravel\Cashier\Exceptions\UnauthorizedInvoiceAccessException
+     */
+    public function findInvoice($orderId)
+    {
+        /** @var Order|null $order */
+        $order = Order::find($orderId);
+
+        if (is_null($order)) {
+            return null;
+        }
+
+        if ($order->owner->isNot($this)) {
+            throw new UnauthorizedInvoiceAccessException;
+        }
+
+        return $order->invoice();
+    }
+
+    /**
+     * Find an invoice or throw a 404 or 403 error.
+     *
+     * @param string $id
+     * @return \Laravel\Cashier\Order\Invoice
+     * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+     */
+    public function findInvoiceOrFail($id)
+    {
+        try {
+            $invoice = $this->findInvoice($id);
+        } catch (UnauthorizedInvoiceAccessException $exception) {
+            throw new AccessDeniedHttpException;
+        }
+
+        if (is_null($invoice)) {
+            throw new NotFoundHttpException;
+        }
+
+        return $invoice;
+    }
+
+    /**
+     * Get the invoice instances for this model.
+     *
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    public function invoices()
+    {
+        return $this->orders->invoices();
+    }
+
+    /**
+     * Create an invoice download response.
+     *
+     * @param $orderId
+     * @param array $data
+     * @param string $view
+     * @param \Dompdf\Options|null $options
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function downloadInvoice($orderId, $data = [], $view = Invoice::DEFAULT_VIEW,  Options $options = null)
+    {
+        /** @var Order $order */
+        $order = $this->orders()->where('id', $orderId)->firstOrFail();
+
+        return $order->invoice()->download($data, $view, $options);
+    }
+
+    /**
+     * Add an invoice item to the customer's upcoming invoice.
+     *
+     * @param string $description
+     * @param int $amount
+     * @param array $overrides
+     * @return OrderItem
+     */
+    public function tab($description, $amount, array $overrides = [])
```

Comment:
Need to check that the user has a valid mandate

Reply:
No. You can add a tab for a user without a valid mandate and when you call `$user->invoice()` the "tab" items will be added to invoice and will be redirect to mollie checkout page

Reply:
Let's include a note guarding the mandate in the docs.